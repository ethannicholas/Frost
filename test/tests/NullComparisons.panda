method main() {
    Console.printLine("Nullable Object: Null")
    def a:Object? := null
    Console.printLine(a == null)
    Console.printLine(a !== null)
    Console.printLine()

    Console.printLine("Nullable Object: Non-Null")
    a := 12
    Console.printLine(a == null)
    Console.printLine(a !== null)
    Console.printLine()

    Console.printLine("Nullable Value: Null")
    def b:Int? := null
    Console.printLine(b == null)
    Console.printLine(b !== null)
    Console.printLine()

    Console.printLine("Nullable Value: Non-Null")
    b := 12
    Console.printLine(b == null)
    Console.printLine(b !== null)
    Console.printLine()

    Console.printLine("Non-Nullable Object")
    def s:String := "Hello!"
    Console.printLine(s == null)
    Console.printLine(s !== null)
    Console.printLine()

    Console.printLine("Non-Nullable Value")
    def n:Int := 12
    Console.printLine(n == null)
    Console.printLine(n !== null)
}
