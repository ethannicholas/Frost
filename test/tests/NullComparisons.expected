Nullable Object: Null
true
false
true
false

Nullable Object: Non-Null
false
true
false
true

Nullable Value: Null
true
false
true
false

Nullable Value: Non-Null
false
true
false
true

Non-Nullable Value
false
true
